ehr/src/org/labkey/ehr/table/DefaultEHRCustomizer.java

```diff
@@ -235,7 +235,7 @@ private void doSharedCustomization(AbstractTableInfo ti)
                     UserSchema us = getEHRStudyUserSchema(ti);
                     if (us != null)
                     {
-                        col.setFk(new QueryForeignKey(ti.getUserSchema(), ti.getContainerFilter(), us, null, "Animal", "Id", "Id"));
+                        col.setFk(new QueryForeignKey(ti.getUserSchema(), ti.getContainerFilter(), us, us.getContainer(), "Animal", "Id", "Id"));
```

Comment:
I understand the concern here, but in my reading of the code and some quick testing, when `lookupContainer` is null, we use the container from the `UserSchema` that's the third argument to the constructor here. So this change is harmless but already matches the existing behavior.

Reply:
That's exactly how I was expecting this to behave, but it was not doing this in practice.  I can try to work up a local repro, but this is an instance of needing a lot of setup/context to get it to behave right.

Reply:
OK. I didn't explore the second argument - presumably it's using that ContainerFilter when it generates SQL (pointing at container A) even though the TableInfo thinks that it's bound to the container B. If you're seeing a real-world problem somewhere that's incentive enough to take the change. It wasn't clear to me from your description if this was hypothetical or not.

Reply:
I'll try to find next week to dust off my local EHR instance. If I get an EHR running it should be relatively easy to repro this. I'll try to get a definitive answer on where it's breaking down.

Reply:
I didn't do a full repro, but I was able to verify that the ContainerFilter being passed in is indeed putting the current container ID's into the generated SQL. That would explain what you're seeing. I think this is more readable and also avoids passing the ContainerFilter.

```suggestion
                        col.setFk(new QueryForeignKey.Builder(us, null).table("Animal").key("Id").display("Id"));
```

There are a bunch of other places that are following the same pattern for other lookups. Many will only be used in the context of the EHR container, but if this suggestion works for you, we should do a wider review.

Reply:
OK, you could well be right. I unfortunately am under some deadlines and spend way too much time in dataintegration this week, but I will get this running locally to actually inspect it. I would believe this isnt the only place we should consider changing.